server/service/appconfig.go

```diff
@@ -332,6 +332,9 @@ func (svc *Service) ModifyAppConfig(ctx context.Context, p []byte, applyOpts fle
 	fleet.ValidateEnabledVulnerabilitiesIntegrations(appConfig.WebhookSettings.VulnerabilitiesWebhook, appConfig.Integrations, invalid)
 	fleet.ValidateEnabledFailingPoliciesIntegrations(appConfig.WebhookSettings.FailingPoliciesWebhook, appConfig.Integrations, invalid)
 	fleet.ValidateEnabledHostStatusIntegrations(appConfig.WebhookSettings.HostStatusWebhook, invalid)
+
+	svc.validateMDM(ctx, appConfig.MDM, invalid)
```

Comment:
This would trigger the team lookup and validation every time the app config is modified, if there is a default team. It would be best to do it only if the request actually changes the value. You can see some examples in this function that compares the original value vs the new one, it can be used to decide if validation should be triggered. And I think the license check could be done before validating too, so that if we end up removing the default team due to the license, we don't validate the team name for nothing.

Reply:
We should have some tests too around that - call with free license + a team ensuring that it doesn't set one, and premium and invalid team/valid team. Probably at the integration level would make sense (there's `TestAppConfig` in `integration_core_test.go` for the free license, and `TestCustomTransparencyURL` in `integration_enterprise_test.go` can serve as inspiration for premium license tests).